src/plugins/projectexplorer/projectvcsstatus.cpp

```diff
@@ -153,6 +162,15 @@ NodeFileList ProjectVcsStatus::changedFileNodesForCurrentProject() const
                 if (folderNode == nullptr) {
                     return false;
                 }
+                if (dynamic_cast<const ResourceEditor::ResourceTopLevelNode *>(folderNode) != nullptr) {
+                    return projectStatusCache.contains(folderPathOf(folderNode->parentFolderNode()));
+                }
+                else if (const auto * resourceFolder = dynamic_cast<const ResourceEditor::ResourceFolderNode *>(folderNode)) {
+                    // TODO (CvW) use ResourceFolderNode::resourceNode instead, only possible in a seperate plugin because of plugin deps.
+                    if (auto * resourceTopLevel = findResourceTopLevelNode(folderNode)) {
```

Comment:
Is this correct? - It seems we extract the resourceTopLevel but are not actually using it.
Is not every ResourceFolderNode a child of some ResourceTopLevelNode?
What is the intended logic of this code?

Reply:
done